guestlink.php

```diff
@@ -110,6 +116,12 @@
     $bbbsession['guest'] = true;
 
     \mod_bigbluebuttonbn\locallib\bigbluebutton::view_bbbsession_set($context, $bbbsession);
+
+    // Groups handling
+    if ($group !== -1) {
```

Comment:
I wonder if we should do some additional validation here. Can someone force a submission with a groupID they dont have access to, and get in?

Reply:
Since there doesn't appear to be a separate guest link generated per group (e.g. with a different url, access code, etc) it's using a single guest link with a group designation. Since these group ids are typically sequential in nature, it's easy to jump around if you know what you are doing.

A better solution might be to have a separate guest link available per group. This however might limit the possibility of a guest joining a different session if that was infact intended at any given point, and would mean the data would need to probably live in a separate table (one to many) since the groups and linking are generated based on base meeting id, and moodle's group id. Alternatively, since they are not both stored in the bigbluebuttonbn table, it might be necessary to add a `group` field to this table to allow for it to be unique and can be directly referenced.

tl;dr: It's a bit weak yes, depends if you think by design it should work one way or the other, or be configurable. Also, this would require a DB table change of some sort.

Reply:
Id like to do a permissions check then, to see if the user can actually access this group. There should be a function in grouplib.php hopefully. Lets just spend a little bit of time to see if we can harden it a little